### ISSUE TITLE
StyleT2I: Toward Compositional and High-Fidelity Text-to-Image Synthesis

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#173]
## TL; DR

- Leverage pretrained *foundational model (CLIP)* and *generative model (StyleGAN2)* for text-to-image synthesis.
- Explicitly strengthen the synthesis compositionality with contrastive loss formulation.

## Motivation

- Lift unconditional generative model for text-conditioned image synthesis.
- Make model robust and fair towards under-represented attribute combinations in the dataset.

<img width="714" alt="image" src="https://user-images.githubusercontent.com/46308407/165891427-13e66602-5fb0-42d1-aa70-a43af28d9caa.png">

### Text-to-Direction

- *Text-to-direction module* takes in (latent code, text) and outputs offset in the latent space that alters the given latent code to produce text-aligned image.
- Fine-tuned CLIP provides embedding similarity signals, and l2 normalization prevents the altered latent code to escape the original latent space (support)

<img width="712" alt="image" src="https://user-images.githubusercontent.com/46308407/165891690-52aa2d80-76da-4fa1-ac8c-d0c750a1a958.png">
<img width="698" alt="image" src="https://user-images.githubusercontent.com/46308407/165891703-4367a016-e473-41fc-805c-9abef674c53c.png">
<img width="337" alt="image" src="https://user-images.githubusercontent.com/46308407/165891714-e79547ec-4c06-4e24-aaf4-0f7c9c908938.png">

### Attribute-to-Direction

- Extract attributes (pre-defined) from text prompts and compose an *attribute vocabulary*.
- Add and subtract the attribute embedding (=output of the *attribute-to-direction module*) from an original random latent vector, and enforce such that CLIP embedding distance between negative and text is large and that between positive and text is small, as follows.

<img width="703" alt="image" src="https://user-images.githubusercontent.com/46308407/165936693-56198e1c-d6c3-4455-aab0-9c0e0d44309d.png">
<img width="589" alt="image" src="https://user-images.githubusercontent.com/46308407/165936726-a0ef120d-934c-4c0a-96cd-d85b49f9766b.png">

- As this loss term alone can result in ***entangled*** modification (i.e., not altering the attribute in isolated manner) additionally employ spatial constraint (to keep irrelevant part fixed).
- Off-the-shelf part segmentation network is used to generate pseudo label, and pixel difference in outer region incurs penalty.

<img width="715" alt="image" src="https://user-images.githubusercontent.com/46308407/165936917-f8a5f94a-ef5c-4e70-9fc8-d5a186619451.png">
<img width="344" alt="image" src="https://user-images.githubusercontent.com/46308407/165936933-99bd1185-973c-49ac-9ce8-63922222d9db.png">

### Compositional Text-to-Image

- Leverage the two modules (text-to-direction, attribute-to-direction) to generate images that incorporates ***all*** components of the text prompt.
- First generate text direction.
- Then, for each attribute that is not entailed in the text direction, explicitly add them as normalized offsets.

<img width="628" alt="image" src="https://user-images.githubusercontent.com/46308407/165937362-f8e4ca60-c0fc-4ae4-a911-6c280b436250.png">

## Evaluation

<img width="706" alt="image" src="https://user-images.githubusercontent.com/46308407/165937465-b0842d99-2333-44e3-8565-90cb58fbfda1.png">
<img width="704" alt="image" src="https://user-images.githubusercontent.com/46308407/165937487-23a1c229-30df-4eb7-8062-72d22c6f5657.png">
<img width="1447" alt="image" src="https://user-images.githubusercontent.com/46308407/165937510-53fcd71f-8943-4468-a224-92c8b1b2676b.png">
<img width="705" alt="image" src="https://user-images.githubusercontent.com/46308407/165937523-936850af-39a6-4f04-a98a-fafb271950d4.png">
<img width="716" alt="image" src="https://user-images.githubusercontent.com/46308407/165937537-b6c1aa35-2978-42fa-8a2a-83a35c4bb0ff.png">
<img width="711" alt="image" src="https://user-images.githubusercontent.com/46308407/165937552-91378b84-d29f-4ac2-82c2-96b6ca3e7213.png">
<img width="716" alt="image" src="https://user-images.githubusercontent.com/46308407/165937570-a3e1fd10-80b2-4054-9852-2e76cd55917f.png">

## Inspiration

- Compositionality: taking care of *under-represented* manifold and mitigating dataset bias
- Fine-tune last layers of CLIP for detailed guidance
